public/app/features/alerting/unified/testSetup/datasources.ts

```diff
@@ -2,15 +2,21 @@ import { keyBy } from 'lodash';
 
 import { DataSourceInstanceSettings } from '@grafana/data';
 import { config, setDataSourceSrv } from '@grafana/runtime';
-
-import { MockDataSourceSrv } from '../mocks';
+import { DatasourceSrv } from 'app/features/plugins/datasource_srv';
 
 /**
  * Sets up the data sources for the tests.
  * Sets up both config object from grafana/runtime and the data source server
  * @param configs data source instance settings. Use **mockDataSource** to create mock settings
  */
 export function setupDataSources(...configs: DataSourceInstanceSettings[]) {
-  config.datasources = keyBy(configs, (c) => c.name);
-  setDataSourceSrv(new MockDataSourceSrv(config.datasources));
+  const dataSourceSrv = new DatasourceSrv();
+  const datasourceSettings = keyBy(configs, (c) => c.name);
+
+  const defaultDatasource = configs.find((c) => c.isDefault);
+  config.datasources = datasourceSettings;
+  dataSourceSrv.init(config.datasources, defaultDatasource?.name || config.defaultDatasource);
+  setDataSourceSrv(dataSourceSrv);
```

Comment:
This is the main change in the PR - it's more accurate to what the datasource service actually does, as it now uses the real service rather than our mock. This means they get "setup" properly and try and load their corresponding query editors etc in our tests